### ISSUE TITLE
Add command entry autocomplete

### ISSUE THREAD

**Issue by andrewtavis** [scribe-org/Scribe-iOS#299]
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/scribe-org/Scribe-iOS/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow Scribe-iOS' [Code of Conduct](https://github.com/scribe-org/Scribe-iOS/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

Now that the data solution has been improved in #96, Scribe has dramatically more memory available to do some things that would have been difficult before. One great feature would be to add autocomplete to the command bar such that a user would be able to press to complete the rest of a word that's alphabetically the next word in the index column of the table being queried. Things to consider are:

- We need to have the remaining text be greyed out to indicate that the autocomplete is possible
- We cannot use the spacebar for this as there could be reflexive verbs in the future (to apple in German is `sich bewerben`) or small phrases
- We need to figure out some kind of indicator to show that user that autocomplete is possible

With this in mind, to me the most obvious place for the user to click would be on the command bar itself. We could also show a similarly greyed out icon on the right side of the command bar to indicate that the bar can be clicked to enter the autocomplete text. A good option for that would be `hand.tap` from SF Symbols, which can be seen in the mockup below:

![Screenshot_2023-04-30 00 15 40](https://user-images.githubusercontent.com/24387426/235326314-93d5193b-d6ff-4a75-ad24-a4c233bb2eda.png)

### Contribution

Happy to work on this myself or collaborate with someone on this :) I think this would be another really great feature to add!

**Comment by andrewtavis** [comment]
@SaurabhJamadagni, would be great to get your feedback on this! I think it could really improve the user experience as it would also mitigate how much people would enter in something wrong 🤔 Would be very grateful to hear if you think this is the best way to do this though 😊

**Comment by andrewtavis** [comment]
Suggestions for how best to implement this would also be welcome :D

**Comment by MykytaBabanin** [comment]
Hello, @andrewtavis  I'd like to take this issue, so could you please assign it to me. It will be my first issue on this project.

**Comment by andrewtavis** [comment]
Hello @MykytaBabanin! 👋 Great to have your interest in Scribe! Happy to assign you :)

As of now Scribe actually doesn’t have an English keyboard, but we’ll be adding one in #7 once #16 is done. This feature would be available in all current keyboards though. Generally my thinking is that each time the user types in a command there is a new query of the corresponding SQLite database that returns the next word alphabetically (`LIMIT 1` at the end). So a user presses the conjugate button after pressing the Scribe key and they’re asked to enter a verb to conjugate. As they type the word updates and they can press the icon on the side of the command bar or maybe the command bar itself to complete it. 

Happy to answer more questions about where to look for things! This should be a fun issue and you’ll learn a lot about how the app works 😊

**Comment by SaurabhJamadagni** [comment]
Hey @andrewtavis, this sounds like a reasonable solution. My only concern with this is unintentional keyboard clicks when trying to tap the command bar. I think this would be a good solution. What about long pressing the spacebar? Could that be an option if we can't have just clicking the spacebar? The icon idea is great as well. 😄

**Comment by andrewtavis** [comment]
Maybe let's go with just clicking the icon and not the command bar itself? We can also reuse some of the mechanics from `commandState = .invalid` in that case :) I'm thinking that getting someone to know that they need to long hold the spacebar is something that would be too difficult to do without a full tutorial.

**Comment by andrewtavis** [comment]
@MykytaBabanin, checking in with you to see if there’s anything we can do to support you here. This is definitely a tough issue, so let us know if you need parts of the codebase explained or if we should break it down into smaller steps :)

**Comment by SaurabhJamadagni** [comment]
Yes @andrewtavis, I would be willing to assist @MykytaBabanin as well! Please let me know if there is any assistance I can provide :)
